test/Modules/UngradedModules.jl

```diff
@@ -0,0 +1,675 @@
+> with exponents between `0` and `max_exp` und between `0` and `max_terms` terms
+"""
+function randpoly(R::Oscar.Ring,coeffs=0:9,max_exp=4,max_terms=8)
+	n = nvars(R)
+	K = base_ring(R)
+	E = [[Random.rand(RNG,0:max_exp) for i=1:n] for j=1:max_terms]
+	C = [K(Random.rand(RNG,coeffs)) for i=1:max_terms]
+	M = MPolyBuildCtx(R)
+	for i=1:max_terms
+		push_term!(M,C[i],E[i])
+	end
+	return finish(M)
+end
+
+"""
+	array_to_matrix(A::Array,R::Ring = parent(A[1,1]))
+Return `A` as an AbstractAlgebra Matrix
+"""
+function array_to_matrix(A::Array,R::AbstractAlgebra.Ring = parent(A[1,1]))
+	Mat = AbstractAlgebra.MatrixSpace(R,size(A)...)
+	return Mat(R.(A))
+end
+
+@testset "Intersection of modules" begin
+  R, (x, y, z) = PolynomialRing(QQ, ["x", "y", "z"])
+
+  A1 = R[x y;
+        2*x^2 3*y^2]
+  A2 = R[x^3 x^2*y;
+        (2*x^2+x) (2*y^3+y)]
+  B = R[4*x*y^3 (2*x+y)]
+  F2 = Oscar.FreeMod(R,2)
+
+  M1 = Oscar.SubQuo(F2, A1, B)
+  M2 = Oscar.SubQuo(F2, A2, B)
+
+  A1 = R[x R(1); y x^2]
+  A2 = R[x y]
+  B1 = R[x^2 y^2]
+  res = R[(x*y^2) (y^3-x*y+y^2); (-x*y) (-x^2*y^2+x*y^3-x^2*y+y^3-x*y)]
+
+  @test M1 == intersect(M1, M1)[1]
+  @test Oscar.SubQuo(F2, res, B1) == intersect(Oscar.SubQuo(F2, A1, B1), Oscar.SubQuo(F2, A2, B1))[1] #macaulay2
+
+  A1 = R[x y; x^2 y^2]
+  A2 = R[x+x^2 y+y^2]
+  @test Oscar.SubQuo(F2, A2, B1) == intersect(Oscar.SubQuo(F2, A1,B1), Oscar.SubQuo(F2, A2,B1))[1]
+end
+
+@testset "Presentation" begin
+
+	# over Integers
+	#=R, (x,y,z) = PolynomialRing(ZZ, ["x", "y", "z"])
+	generator_matrices = [R[x x^2*y; y y^2*x^2], R[x x^2*y; y y^2*x^2], R[x y; x^2*y y^2*x], R[x+R(1) x^10; x^2+y^2 y^4-x^4], R[42*x*y 7*x^2; 6*x 9*y^2]]
+	relation_matrices  = [R[x^3 y^4], R[x^3 y^4; x^2*y x*y^2], R[x*y^2 x; y^3 x*y^3], R[x x*y], R[3*x*y 7*x*y; 42 7]]
+	true_pres_matrices = [R[x^5*y-y^4 -x^5+x*y^3], R[-x^2*y-x*y-y x^2+x; x*y^4-x^3*y+y^4-x^2*y -x*y^3+x^3; -y^4+x^2*y x*y^3-x^3; x^2*y^3+x*y^3+y^3 -x^2*y^2-x*y^2], R[-x*y R(1); -x^2*y^5+x*y^3 R(0)], R[x^5-x*y^4+x^3*y+x*y^3 x^11-x^2*y-x*y; -x^5*y^7+x*y^11+2*x^5*y^6-x^3*y^8-3*x*y^10+2*x^5*y^5+2*x^3*y^7-3*x^5*y^4+2*x^3*y^6+5*x*y^8+2*x^5*y^3-3*x^3*y^5-5*x*y^7+2*x^3*y^4+2*x*y^6 -x^11*y^7+2*x^11*y^6+2*x^11*y^5-3*x^11*y^4+2*x^11*y^3+x^2*y^8-2*x^2*y^7+x*y^8-2*x^2*y^6-2*x*y^7+3*x^2*y^5-2*x*y^6-2*x^2*y^4+3*x*y^5-2*x*y^4], R[-13*y R(0); -377 -2639*x; -39 -273*x; -13*y-39 -273*x; -13 -91*x; y^2-42*x -294*x^2+21*x*y; 9*y^2-x+26*y+78 -7*x^2+189*x*y+546*x; -y^2+3*x 21*x^2-21*x*y]]
+	for (A,B,true_pres_mat) in zip(generator_matrices, relation_matrices, true_pres_matrices)
+    println(A,B,true_pres_mat)
+		SQ = Oscar.SubQuo(A,B)
+		pres_mat = matrix(Oscar.present_as_cokernel(SQ).quo)
+		@test Oscar.cokernel(pres_mat) == Oscar.cokernel(true_pres_mat)
+
+		pres_SQ, i = Oscar.present_as_cokernel(SQ, :both)
+		p = i.inverse_isomorphism
+		@test Oscar.iswelldefined(i)
+		@test Oscar.iswelldefined(p)
+		@test Oscar.isbijective(i)
+		@test Oscar.isbijective(p)
+	end=#
+
```

Comment:
Why is this commented out? I'd suggest to either add a comment explaining the reason (so that we may know when to try reactivating it), or else remove it?